qiita_pet/handlers/study_handlers/artifact.py

```diff
@@ -21,3 +25,72 @@ def get(self):
         artifact = to_int(self.get_argument('artifact_id'))
         self.write(artifact_graph_get_req(artifact, direction,
                                           self.current_user.id))
+
+
+class ArtifactAJAX(BaseHandler):
+    def get(self):
+        artifact_id = to_int(self.get_argument('artifact_id'))
+        self.write('TITLE FOR ARTIFACT %d' % artifact_id)
+
+    def post(self):
+        artifact_id = to_int(self.get_argument('artifact_id'))
+        self.write(artifact_delete_req(artifact_id, self.current_user.id))
+
+
+class ArtifactAdminAJAX(BaseHandler):
+    def get(self):
```

Comment:
How do others feel about creating all those HTML buttons in here?
I think it is better the previous approach that was used. The buttons are already defined in the interface and we just pass booleans to the template to decide if a buttons is shown or not. The approach taken here, splits the definition of the template between the HTML and the handler which, in my opinion, makes it harder to maintain.